1. No issue ref?
2. Test failure

```
--- FAIL: TestTemporaryUploadRepository (0.00s)
    temp_repo_test.go:22: 
        	Error Trace:	/home/runner/work/gitea/gitea/services/repository/files/temp_repo_test.go:22
        	Error:      	Received unexpected error:
        	            	invalid object format: sha256
        	Test:       	TestTemporaryUploadRepository
FAIL

--- a/services/repository/files/temp_repo_test.go
+++ b/services/repository/files/temp_repo_test.go
@@ -15,17 +15,31 @@ import (
 
 func TestTemporaryUploadRepository(t *testing.T) {
 	mockedRepo := &repo_model.Repository{Name: "mocked-repo-name", OwnerName: "mocked-owner-name"}
-	tmpGitRepo, err := NewTemporaryUploadRepository(mockedRepo)
-	require.NoError(t, err)
-	defer tmpGitRepo.Close()
 
-	require.NoError(t, tmpGitRepo.Init(t.Context(), git.Sha256ObjectFormat.Name()))
+	doTest := func(t *testing.T, objectFormatName string) {
+		tmpGitRepo, err := NewTemporaryUploadRepository(mockedRepo)
+		require.NoError(t, err)
+		defer tmpGitRepo.Close()
 
-	require.NoError(t, tmpGitRepo.RemoveFilesFromIndex(t.Context(), "any-file-name"))
-	require.NoError(t, tmpGitRepo.RemoveFilesFromIndex(t.Context(), "--any-file-name"))
+		require.NoError(t, tmpGitRepo.Init(t.Context(), objectFormatName))
 
-	objID, err := tmpGitRepo.HashObjectAndWrite(t.Context(), bytes.NewReader(nil))
-	require.NoError(t, err)
-	require.NoError(t, tmpGitRepo.AddObjectToIndex(t.Context(), "100644", objID, "any-file-name"))
-	require.NoError(t, tmpGitRepo.AddObjectToIndex(t.Context(), "100644", objID, "--any-file-name"))
+		require.NoError(t, tmpGitRepo.RemoveFilesFromIndex(t.Context(), "any-file-name"))
+		require.NoError(t, tmpGitRepo.RemoveFilesFromIndex(t.Context(), "--any-file-name"))
+
+		objID, err := tmpGitRepo.HashObjectAndWrite(t.Context(), bytes.NewReader(nil))
+		require.NoError(t, err)
+		require.NoError(t, tmpGitRepo.AddObjectToIndex(t.Context(), "100644", objID, "any-file-name"))
+		require.NoError(t, tmpGitRepo.AddObjectToIndex(t.Context(), "100644", objID, "--any-file-name"))
+	}
+
+	t.Run("sha1", func(t *testing.T) {
+		doTest(t, git.Sha1ObjectFormat.Name())
+	})
+
+	t.Run("sha256", func(t *testing.T) {
+		if !git.DefaultFeatures().SupportHashSha256 {
+			t.Skip("sha256 is not supported")
+		}
+		doTest(t, git.Sha256ObjectFormat.Name())
+	})
 }

